elk/extraction/prompt_dataset.py

```diff
@@ -89,8 +89,10 @@ def __init__(self, cfg: PromptConfig):
         self.strategy = cfg.strategy
 
         # TODO: Should we support IterableDataset?
-        ds_dict = load_dataset(*data_path)  # type: ignore
-        assert isinstance(ds_dict, DatasetDict)
+        _ds_dict = load_dataset(*data_path)  # type: ignore
+        assert isinstance(_ds_dict, DatasetDict)
+        ds_dict: DatasetDict = _ds_dict
```

Comment:
unfortunately mypy needs this "dance"

Reply:
Yep this seems sufficiently ugly that I'm not sure I want it in the code. Is there really no cleaner way of doing the same thing?

Reply:
see https://github.com/EleutherAI/elk/pull/93/files/245859cb3d5a74e569e0800c191ee7ef4a02feb7..51db6995fe583d29ae5e3a863fda75724af6ce76#r1113128336